lib/trie/hash.go

```diff
@@ -17,31 +17,64 @@
 package trie
 
 import (
+	"bytes"
+	"context"
 	"hash"
+	"sync"
+
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/ChainSafe/gossamer/lib/scale"
+	"golang.org/x/sync/errgroup"
 
 	"golang.org/x/crypto/blake2b"
 )
 
+type sliceBuffer []byte
+
+func (b *sliceBuffer) Write(data []byte) (n int, err error) {
+	*b = append(*b, data...)
+	return len(data), nil
+}
+
+func (b *sliceBuffer) Reset() {
+	*b = (*b)[:0]
+}
+
```

Comment:
do these funcs need to be exported? doesn't look like they need to be

Reply:
Done